rpc/src/transaction_status_service.rs

```diff
@@ -43,33 +43,44 @@ impl TransactionStatusService {
                         break;
                     }
 
-                    let message = match write_transaction_status_receiver
-                        .recv_timeout(Duration::from_secs(1))
-                    {
+                    let message = match write_transaction_status_receiver.try_recv() {
                         Ok(message) => message,
-                        Err(RecvTimeoutError::Disconnected) => {
+                        Err(TryRecvError::Disconnected) => {
                             break;
                         }
-                        Err(RecvTimeoutError::Timeout) => {
+                        Err(TryRecvError::Empty) => {
+                            // TSS is bandwidth sensitive at high TPS, but not necessarily
+                            // latency sensitive. We use a global thread pool to handle
+                            // bursts of work below. This sleep is intended to balance that
+                            // out so other users of the pool can make progress while TSS
+                            // builds up a backlog for the next burst.
+                            sleep(Duration::from_millis(50));
```

Comment:
50ms seems like an awful long time. Forgive me if this has already been discussed, but why not, say, 10ms?

Reply:
Fair enough. I'm cool with moving forward with this value - can always tune it later